docs/specifications/tests/Consistency-TP/consistency05.md

```diff
@@ -1,49 +1,200 @@
+
+   1. Extract the [in-bailiwick] name servers names and create a set
+      "Delegation Strict Glue" of those, were each name server name 
+      is matched with its IP address or addresses, if available. The 
+      set might be empty.
+   2. Extract the [out-of-bailiwick] name servers names and create a set
+      "Delegation Extended Glue" of those, were each name server name 
+      is matched with its IP address or addresses, if available. The 
+      set might be empty.
+
+2. Obtain the IP addresses of the [in-bailiwick] name servers from the
+   *Child Zone*:
+
+   1. Obtain the set of name server names for the *Child Zone* using
+      [Method2] and [Method3].
+   2. Extract the [in-bailiwick] name servers. If there are no such 
+      name servers, skip the steps to extract IP address of the 
+      [in-bailiwick] name servers from *Child Zone*.
+   3. For each [in-bailiwick] name server name create one A query and 
+      one AAAA query with the RD flag unset.
+   4. Obtain the set of "NS IP" for *Child Zone* from [Method4] and
+      [Method5].
+   5. Repeat the steps below for each name server in *NS IP* and for
+      each address (A, AAAA) query.
+   6. Send the address query to the name server.
+   7. If there is no DNS response from the server then:
+      1. Emit *[NO_RESPONSE]*.
+      2. Go to the next server.
+   8. If the response is a delegation (referral) to a sub-zone of 
+      *Child Zone*, then:
+      1. Copy the adress query (A, AAAA) that gave the referral
+         response.
+      2. Set the RD flag in the copied query (from unset to set).
+      3. Do a DNS lookup of the the query using a resolving name server 
+         on the public DNS.
+         * The lookup must take into account changes that
+           undelegated data has created, if any.
+      4. If the lookup returns the relevant address record or records
+         with the same owner name as in the query, then extract those 
+         to be collected below.
+      5. Go to the next server.
+   9. If the response has the AA flag unset, then:
+      1. Emit *[CHILD_NS_FAILED]*. 
+      2. Go to the next server.
+   10. If the RCODE of the response is neither NOERROR nor NXDOMAIN, 
+       then:
+       1. Emit *[CHILD_NS_FAILED]*.
+       2. Go to the next server.
+   11. If the RCODE of the response is NXDOMAIN, then go to the next 
+       server.
+   12. If the RCODE is NOERROR (with the AA flag set), then:
+       1. Extract any addresses records (A, AAAA) from the answer
+          section response if the owner name matches the that of 
+          the of the query.
+       2. Go to the next server.
+   13. When all servers are processed, then go to the next step.
+   14. If all servers emitted *[NO_RESPONSE]* or 
+       *[CHILD_NS_FAILED]*, then emit *[CHILD_ZONE_LAME]* and
+       completely stop processing this test case.
+   15. Tag any extracted address records (A and AAAA) as 
+       "Address Records From Child".
```

Comment:
The approach employed here is to start by extracting some records here and some records there, leaving them floating in the air for a while until this step where they're all gathered up and placed in a named collection.

I would like to suggest another approach: Start with creating an empty named collection, possibly with a note that records will be added to it in following steps. Then add records to the named collection as they are extracted.

This way each record has its clear place and context at every instant. This is much easier on the reader.

Reply:
Updated.